python/python/lance/torch/kmeans.py

```diff
@@ -43,6 +51,8 @@ class KMeans:
     device: str, optional
         The device to run the PyTorch algorithms. Default we will pick
         the most performant device on the host. See `lance.torch.preferred_device()`
+    use_cuvs: bool
```

Comment:
can we detect this automatically? i.e., `devcie == cuda AND num_partitons > blablabla`

Reply:
Unfortunately this adds another dataset-dependent parameter with no obvious threshold/choice.
I set up a new "accelerator" type for this. We could add automatic accelerator choices in the future though (just as decent untuned defaults).